awsbestpractices/batch-write-output-ignored/BatchWriteOutputIgnoredRule.java

```diff
@@ -0,0 +1,123 @@
+import com.amazonaws.services.glue.model.BatchDeleteTableVersionRequest;
+import com.amazonaws.services.glue.model.BatchDeleteTableVersionResult;
+
+public class Example {
+    private AWSGlueClient client;
+
+    // Must be flagged
+    public void ignoreOutput_1(final String databaseName, final List<String> tableNames) {
+        client.batchDeleteTable(new BatchDeleteTableRequest().withDatabaseName(databaseName)
+                .withTablesToDelete(tableNames));
+    }
+
+    // Must be flagged.
+    // Based on
+    // https://code.amazon.com/packages/IronmanAdminService/blobs/aa325ce9f79c445e5ff51b46cddf9259b97e7a9c/--//src/com/amazonaws/sagemaker/admin/activity/operational_support/DeadLetterSqsMessages.java#L124
+    public void ignoreOutput_2(final String accountId,
+                               final String queueName,
+                               List<Message> messages) {
+        AmazonSQS sqsClient = clientProvider.getSqsClientForRead(accountId);
+        String dlqUrl = getQueueUrl(sqsClient, accountId, queueName).getDlq();
+
+        List<DeleteMessageBatchRequestEntry> deleteMessages = messages.stream()
+                .map(x -> new DeleteMessageBatchRequestEntry()
+                        .withId(x.getMessageId())
+                        .withReceiptHandle(x.getReceiptHandle()))
+                .collect(Collectors.toList());
+
+        sqsClient.deleteMessageBatch(dlqUrl, deleteMessages);
+    }
+
+    // Must be flagged.
+    // Based on
+    // https://code.amazon.com/packages/IronmanSearchService/blobs/8acfee2d8d34459c762b4fa4ee525cbf76aaa2b4/--//src/com/amazon/ironmansearchservice/activity/ProcessS3EventDlqMessagesActivity.java#L95
+    private List<String> ignoreOutput_3(final int numberOfMessages, AmazonSQS amazonSQS) {
+        List<String> messages = new ArrayList<>();
+        if (!StringUtils.isBlank(queueURL)) {
+            try {
+                int batchSteps = 1;
+                int maxForBatch = numberOfMessages;
+                if ((numberOfMessages > MAX_MESSAGES)) {
+                    batchSteps = (numberOfMessages / MAX_MESSAGES) + ((numberOfMessages % MAX_MESSAGES > 0) ? 1 : 0);
+                    maxForBatch = MAX_MESSAGES;
+                }
+
+                for (int i = 0; i < batchSteps; i++) {
+                    List<Message> list = amazonSQS.receiveMessage(new ReceiveMessageRequest(queueURL).
+                            withMaxNumberOfMessages(maxForBatch).withWaitTimeSeconds(POLLING_INTERVAL)).getMessages();
+                    if (list != null && !list.isEmpty()) {
+                        List<DeleteMessageBatchRequestEntry> del = new ArrayList<>();
+                        for (Message msg : list) {
+                            messages.add(msg.getBody());
+                            del.add(new DeleteMessageBatchRequestEntry(msg.getMessageId(), msg.getReceiptHandle()));
+                        }
+                        amazonSQS.deleteMessageBatch(queueURL, del);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code makes a call to a batch operation, but it does not handle any errors returned by that batch operation. Batch operations can succeed without throwing exceptions even if some items in the request fail. Consider checking the return value of the batch operation for failures.